### PR TITLE
feat: Bring in latest `NavigationView` changes from MUX

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #

Based on https://github.com/microsoft/microsoft-ui-xaml/pull/8776

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Feature

## What is the current behavior?

Quite a few changes are missing in Uno.

## What is the new behavior?

- Changes added and commit references bumped.
- File-scoping

### <samp>🤖 Generated by Copilot at 3232e6b</samp>

This pull request improves the navigation view control and its related classes by updating the MUX reference files, using the new file-scoped namespace syntax, adding setters for the chevron icon color, fixing a binding typo, adding a source comment, optimizing the pane header loading, simplifying the UI automation code, adding a new test method, and separating the header and non-header dependency properties into different files.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewTemplateSettings.Properties.cs [hunk 1]
```diff
@@ -1,173 +1,172 @@
...
+	public static DependencyProperty LeftPaneVisibilityProperty { get; } =
+		DependencyProperty.Register(nameof(LeftPaneVisibility), typeof(Visibility), typeof(NavigationViewTemplateSettings), new FrameworkPropertyMetadata(Visibility.Visible));
+
+	/// <summary>
+	/// Gets the width of open pane.
+	/// </summary>
+	public double OpenPaneLength
+	{
+		get => (double)GetValue(OpenPaneLengthProperty);
+		set => SetValue(OpenPaneLengthProperty, value);
```
Comment (jeromelaban) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewTemplateSettings.Properties.cs:58]: Won't this be a breaking change?

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewTemplateSettings.Properties.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewTemplateSettings.Properties.cs:58]: The removal of `OpenPaneWidth` is a breaking change, but the `OpenPaneLength` property was there already generated by Sync Gen. WinUI did the rename when the API became stable.

**Comment by MartinZikmund** [comment]
Need help from @dr1rrb here, as a test is failing due to a pointer-related problem and I am not sure why that's the case...

**Comment by MartinZikmund** [comment]
@dr1rrb Ramez was able to fix the issue, so no further action needed for now 👍

**Pull request merged**